File: sdks/typescript/src/adapters/autogen.ts
import { NexusClient } from "../client";

export async function recordAutoGenMessage(
  client: NexusClient,
  input: { agentId: string; runId: string; channel: string; sender: string; content: string },
): Promise<void> {
  await client.publishDelta({
    channel: input.channel,
    agentId: input.agentId,
    runId: input.runId,
    delta: { sender: input.sender, content: input.content },
    durable: true,
    tags: ["autogen", "message"],
  });
}

File: sdks/python/nexus_ipc/adapters/autogen.py
from __future__ import annotations

from typing import Any

from nexus_ipc.client import NexusClient


def record_message(
    client: NexusClient,
    agent_id: str,
    run_id: str,
    channel: str,
    sender: str,
    content: str,
) -> None:
    client.publish_delta(
        channel=channel,
        agent_id=agent_id,
        run_id=run_id,
        delta={"sender": sender, "content": content},
        durable=True,
        tags=["autogen", "message"],
    )


def middleware_event(client: NexusClient, agent_id: str, run_id: str, event: dict[str, Any]) -> None:
    client.checkpoint(
        agent_id=agent_id,
        run_id=run_id,
        channel="topic:autogen_events",
        state=event,
        summary="AutoGen middleware event",
        tags=["autogen"],
    )

